code/modules/surgery/bodyparts/_bodyparts.dm

```diff
@@ -784,42 +798,94 @@
 	drop_organs()
 	return ..()
 
-/**
- * Calculates how much blood this limb is losing per life tick
- *
- * Arguments:
- * * ignore_modifiers - If TRUE, ignore the bleed reduction for laying down and grabbing your limb
- */
-/obj/item/bodypart/proc/get_part_bleed_rate(ignore_modifiers = FALSE)
+/// INTERNAL PROC, DO NOT USE
+/// Properly sets us up to manage an inserted embeded object
+/obj/item/bodypart/proc/_embed_object(obj/item/embed)
+	if(embed in embedded_objects) // go away
+		return
+	// We don't need to do anything with projectile embedding, because it will never reach this point
+	RegisterSignal(embed, COMSIG_ITEM_EMBEDDING_UPDATE, .proc/embedded_object_changed)
+	embedded_objects += embed
+	refresh_bleed_rate()
+
+/// INTERNAL PROC, DO NOT USE
+/// Cleans up any attachment we have to the embedded object, removes it from our list
+/obj/item/bodypart/proc/_unembed_object(obj/item/unembed)
+	UnregisterSignal(unembed, COMSIG_ITEM_EMBEDDING_UPDATE)
+	embedded_objects -= unembed
+	refresh_bleed_rate()
+
+/obj/item/bodypart/proc/embedded_object_changed(obj/item/embedded_source)
+	SIGNAL_HANDLER
+	/// Embedded objects effect bleed rate, gotta refresh lads
+	refresh_bleed_rate()
+
+/// Sets our generic bleedstacks
+/obj/item/bodypart/proc/setBleedStacks(set_to)
+	adjustBleedStacks(set_to - generic_bleedstacks)
+
+/// Modifies our generic bleedstacks. You must use this to change the variable
+/// Takes the amount to adjust by, and the lowest amount we're allowed to have post adjust
+/obj/item/bodypart/proc/adjustBleedStacks(adjust_by, minimum = -INFINITY)
+	if(!adjust_by)
+		return
+	var/old_bleedstacks = generic_bleedstacks
+	generic_bleedstacks = max(generic_bleedstacks + adjust_by, minimum)
+
+	// If we've started or stopped bleeding, we need to refresh our bleed rate
+	if((old_bleedstacks <= 0 && generic_bleedstacks > 0) \
+		|| old_bleedstacks > 0 && generic_bleedstacks <= 0)
+		refresh_bleed_rate()
+
+/obj/item/bodypart/proc/on_owner_nobleed_loss(datum/source)
+	SIGNAL_HANDLER
+	refresh_bleed_rate()
+
+/obj/item/bodypart/proc/on_owner_nobleed_gain(datum/source)
+	SIGNAL_HANDLER
+	refresh_bleed_rate()
```

Comment:
```suggestion
/obj/item/bodypart/proc/on_owner_nobleed_loss(datum/source)
	SHOULD_CALL_PARENT(TRUE)
	PROTECTED_PROC(TRUE)
	SIGNAL_HANDLER
	refresh_bleed_rate()

/obj/item/bodypart/proc/on_owner_nobleed_gain(datum/source)
	SHOULD_CALL_PARENT(TRUE)
	PROTECTED_PROC(TRUE)
	SIGNAL_HANDLER
	refresh_bleed_rate()
```

Reply:
I uh, why? is it just to mark out to people that they shouldn't be calling these things manually? I didn't think that happen that often.

I'm now noticing that it's consistent across the file, so I'm ok with it on that merit, but it seems a bit strange to me.

Reply:
Yes. Linters are significantly more effective than comments.

Reply:
I'm not a fan of this pattern, it seems weird to need to lint against people calling signal procs directly, seems like fluff? idk